src/components/Quiz.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import Answers from './Answers';
+import { publish } from './Event';
+import QuestionAnswer from './QuestionAnswer';
+import Score from './Score';
+import TimerContainer from './TimerContainer';
+
+function Quiz(props) {
+  const { quiz } = props;
+  const [questionNumber, setQuestionNumber] = useState(1);
+  const [questions, setQuestions] = useState([]);
+  const nextQuestion = (question) => {
+    if (questionNumber < (Number(quiz.noOfQuestionsVal) + 1)) {
+      setQuestions([...questions, question]);
+      publish(`ResetInterval${quiz.quizId}`, quiz.timerVal);
+      setQuestionNumber(questionNumber + 1);
+    }
+  };
+  const shouldShowQuiz = questionNumber < (Number(quiz.noOfQuestionsVal) + 1);
+  return (
+    <div>
+      {shouldShowQuiz && (
+        <TimerContainer
+          questionNum={questionNumber}
+          quiz={quiz}
+        />
+      )}
+      {shouldShowQuiz && (
+        <QuestionAnswer
+          questionNum={questionNumber}
+          nextQuestion={nextQuestion}
+          quiz={quiz}
+        />
+      )}
+      <Score answers={questions} />
+      {!shouldShowQuiz
+      && (
+        <Answers
+          answers={questions}
```

Comment:
Does not make logical sense.

Reply:
Change 'shouldShowQuiz' to 'showFinalResult' and 'Answers' to 'Results'. We have to show the final results when the quiz ended that's why using this code block.